test/suites/integration/msp/debt-collection.test.ts

```diff
@@ -0,0 +1,250 @@
+
+    const matchedEvents = events.filter((e) =>
+      userApi.events.fileSystem.NewStorageRequest.is(e.event)
+    );
+
+    if (matchedEvents.length !== source.length) {
+      throw new Error(`Expected ${source.length} NewStorageRequest events`);
+    }
+
+    const issuedFileKeys = [];
+    for (const e of matchedEvents) {
+      const newStorageRequestDataBlob =
+        userApi.events.fileSystem.NewStorageRequest.is(e.event) && e.event.data;
+
+      if (!newStorageRequestDataBlob) {
+        throw new Error("Event doesn't match NewStorageRequest type");
+      }
+
+      const result = await mspApi.rpc.storagehubclient.isFileInFileStorage(
+        newStorageRequestDataBlob.fileKey
+      );
+
+      if (!result.isFileFound) {
+        throw new Error(
+          `File not found in storage for ${newStorageRequestDataBlob.location.toHuman()}`
+        );
+      }
+
+      issuedFileKeys.push(newStorageRequestDataBlob.fileKey);
+    }
+
+    // Seal block containing the MSP's transaction response to the storage request
+    await userApi.wait.mspResponseInTxPool();
+    await userApi.sealBlock();
+
+    let mspAcceptedStorageRequestDataBlob: any = undefined;
+    let storageRequestFulfilledDataBlob: any = undefined;
+
+    try {
+      const { event: mspAcceptedStorageRequestEvent } = await userApi.assert.eventPresent(
+        "fileSystem",
+        "MspAcceptedStorageRequest"
+      );
+      mspAcceptedStorageRequestDataBlob =
+        userApi.events.fileSystem.MspAcceptedStorageRequest.is(mspAcceptedStorageRequestEvent) &&
+        mspAcceptedStorageRequestEvent.data;
+    } catch {
+      // Event not found, continue
+    }
+
+    try {
+      const { event: storageRequestFulfilledEvent } = await userApi.assert.eventPresent(
+        "fileSystem",
+        "StorageRequestFulfilled"
+      );
+      storageRequestFulfilledDataBlob =
+        userApi.events.fileSystem.StorageRequestFulfilled.is(storageRequestFulfilledEvent) &&
+        storageRequestFulfilledEvent.data;
+    } catch {
+      // Event not found, continue
+    }
+
+    let acceptedFileKey: string | null = null;
+    // We expect either the MSP accepted the storage request or the storage request was fulfilled
+    if (mspAcceptedStorageRequestDataBlob) {
+      acceptedFileKey = mspAcceptedStorageRequestDataBlob.fileKey.toString();
+    } else if (storageRequestFulfilledDataBlob) {
+      acceptedFileKey = storageRequestFulfilledDataBlob.fileKey.toString();
+    }
+
+    if (!acceptedFileKey) {
+      throw new Error(
+        "Neither MspAcceptedStorageRequest nor StorageRequestFulfilled events were found"
+      );
+    }
+
+    // There is only a single key being accepted since it is the first file key to be processed and there is nothing to batch.
+    strictEqual(
+      issuedFileKeys.some((key) => key.toString() === acceptedFileKey),
+      true
+    );
+
+    // Allow time for the MSP to update the local forest root
+    await sleep(3000);
```

Comment:
is there an associated log line we can wait on?

Reply:
I don't know about that. I agree that we should move away from `sleep`